src/Compilers/CSharp/Portable/Symbols/TypeUnification.cs

```diff
@@ -268,6 +265,11 @@ private static void AddSubstitution(ref MutableTypeMap substitution, TypeParamet
             substitution.Add(tp1, t2);
         }
 
+        private static bool AreTypesAndCustomModifiersEqual(TypeWithAnnotations t1, TypeWithAnnotations t2)
+        {
+            return TypeSymbol.Equals(t1.Type, t2.Type, TypeCompareKind.CLRSignatureCompareOptions) && t1.CustomModifiers.SequenceEqual(t2.CustomModifiers);
```

Comment:
>t1.CustomModifiers.SequenceEqual(t2.CustomModifiers) [](start = 102, length = 52)

Is order meaningful for custom modifiers?

Reply:
It looks like we treat `CustomModifiers` order as significant - for instance in `TypeWithAnnotations.Equals()` and `MemberSignatureComparer.HaveSameCustomModifiers()`.